src/main.cpp

```diff
@@ -2078,19 +2078,9 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
                              REJECT_INVALID, "bad-txns-BIP30");
     }
 
-    unsigned int flags = SCRIPT_VERIFY_P2SH;
+    unsigned int flags = SCRIPT_VERIFY_P2SH | SCRIPT_VERIFY_CHECKLOCKTIMEVERIFY;
```

Comment:
This PR intentionally fixes both #2359, and the stale comments + redundant block version checks in the code selecting DERSIG and CLTV flags. This change is part of the latter fix. I would prefer not to do a separate PR, I think it's clear enough here.